### PR TITLE
OPNET-208: Prefer ipv6 on v6-primary dual stack deployments

### PR DESCRIPTION
**Pull request by cybertron**
In ipv6-primary dual stack clusters we need to pass --prefer-ipv6 to runtimecfg so it knows to select the ipv6 address as the primary node ip. This adds a new network type value for this case and the necessary logic to use it.

**- What I did**

**- How to verify it**

**- Description for the changelog**
Add support for ipv6-primary dual stack clusters.

### CODE REVIEW

File: pkg/operator/render_test.go
Comment (mkowalski) [inline pkg/operator/render_test.go:71]: Not introduced by your change, but as you are touching this file, can we maybe add the following 2 scenarios

* `[]string{ipv4, ipv4}`
* `[]string{ipv6, ipv6}`

with `Error: true`? Just to make sure if someone in the future adds a handler for this, they are forced to fix the test

Comment (cybertron) [reply pkg/operator/render_test.go:71]: Those are both valid cases. You're allowed to have multiple CIDRs per ip version. We could add more test cases to ensure they're handled right though.

**Comment by cybertron** [comment]
I'm not sure why the metal-ipi job has failures, but I can see from the machine-configs in the must-gather that the correct arguments are being passed to runtimecfg, so since the other on-prem platforms passed I think we can call it good.

**Comment by cybertron** [comment]
/assign @cgwalters
per the bot recommendation

File: pkg/operator/render_test.go
Comment (mkowalski) [reply pkg/operator/render_test.go:71]: ACK, thanks for explaining! I always thought that we only allow single subnet per IP stack

File: pkg/operator/render.go
Comment (yuqi-zhang) [inline pkg/operator/render.go:203]: So essentially the ordering is the only thing that matters when determining the primary, right?

**Review by yuqi-zhang — APPROVED** [review]
Seems fine to me logically. Adding a 

/hold

for QE testing. Feel free to remove if you feel like pre-merge testing is not needed

File: pkg/operator/render.go
Comment (cybertron) [reply pkg/operator/render.go:203]: Right. Whatever IP version is first in the list must be primary. We've been forcing that to be ipv4 at install-time, but after this change we can stop doing that.

**Comment by cybertron** [comment]
/hold cancel

I don't think this can be tested until the dependent installer patch merges to allow this configuration, which we don't want to do until this merges to make it work correctly. It's a bit of a chicken-and-egg situation.

**Pull request merged**
